backend/src/gpml/db/browse.sql

```diff
@@ -3,10 +3,15 @@
 SELECT DISTINCT ON (t.topic, (COALESCE(t.json->>'start_date', t.json->>'created'))::timestamptz, (t.json->>'id')::int) t.topic, t.geo_coverage, t.json
   FROM v_topic t
 --~ (when (and (:favorites params) (:user-id params) (:resource-types params)) "JOIN v_stakeholder_association a ON a.stakeholder = :user-id AND a.id = (t.json->>'id')::int AND (a.topic = t.topic OR (a.topic = 'resource' AND t.topic IN (:v*:resource-types)))")
+--~ (when (seq (:tag params)) "JOIN json_array_elements(t.json->'tags') tags ON true JOIN json_each_text(tags) tag ON tag.value = ANY(ARRAY[:v*:tag]::varchar[])")
```

Comment:
doesn't it seem that we are adding more and more logic to the sql files ??? ... I suspect there are ways () to code this logic in plain clojure files and call this functions in the sql file
🤔 

https://www.hugsql.org/#using-expressions

```
-- :name clj-expr-generic-update :! :n
/* :require [clojure.string :as string]
            [hugsql.parameters :refer [identifier-param-quote]] */
update :i:table set
/*~
(string/join ","
  (for [[field _] (:updates params)]
    (str (identifier-param-quote (name field) options)
      " = :v:updates." (name field))))
~*/
where id = :id

```

so perhaps it's time to use something like `/* :require [gpml.hugsql :as gpml-hugsql] */`

Reply:
then we also could have tests to check the final string-sql generated

Reply:
We already have some code like this in `gpml.sql-util`. We could move some of this `sql` into that, and may be rename the module. 

One thing that I've had issues with this setup before was updating these utility functions, and getting the sql query to get updated when calling the hugsql generated Clojure function. Do you know what the right way to make sure the new query is run, when developing the function? @tangrammer

Reply:
great that we already are using `gpml.sql-util` inside our hugsql files 😄 ... no need so far of renaming it!

regarding development issues with hugsql changes, we need to explicitely eval/load the gpml.db.*.clj file that is using it. (eg: in emacs C-c C-k `cider-load-buffer`) 
the problem with (dev/refresh) function is that only reload the clj files that have changed but we only changed the sql files in this case ... so, perhaps someone has already written some IDE script to force reload based on that dev user case ... I can try to find something or even write it in emacs lisp

Reply:
Thanks, that is helpful. IIRC, I did try to eval/load the gpml.db.*.clj file, but the query still seems to run a stale version of the SQL. I shall try it again and report back on it!

Reply:
this simple function could help thus i didn't find an official/simple approach to fix the problem

```clojure
(ns dev 
....
)
(defn refresh* []
  (require 'gpml.db.organisation :reload)
  (require 'gpml.db.action :reload)
;; add all the clj db files here 
  (refresh))

```
`dev>(refresh*)`

Reply:
Updated the PR extracting the SQL snippet into a clojure utility function with some basic tests. Could you take another look, @tangrammer ?